### PR TITLE
[suppy] migration from AndroidpublisherV2 to AndroidpublisherV3

### LINKED ISSUE #14271: Supply - download screenshots
I am wondering about screenshots for Android. Currently `supply` only downloads `IMAGES_TYPES` from **[supply.rb](https://github.com/fastlane/fastlane/blob/master/supply/lib/supply.rb#L18)**

If I use/extend the same code for `supply.rb/SCREENSHOT_TYPES`, it's downloading screenshots too! (Need to implement some more logic to name the screenshots)

`SCREENSHOT_TYPES` downloaded are compressed as well.

screen shot 2019-02-19 at 4 42 44 pm
(^ testing code)

However, I've noticed this [comment](https://github.com/fastlane/fastlane/blob/master/supply/lib/supply/setup.rb#L46) - 
```ruby
# We cannot download existing screenshots as they are compressed
# But we can at least download the images
```

What do you think about downloading screenshots as well? I mean, whatever is downloaded is compressed anyway :)

### LINKED ISSUE #14573: [supply] Update to Google Play Developer API v3 (v2 end of life Dec 2019)
Google Play Developer Developer Publishing API v2 (currently used by supply) will end of life Dec 2019.

https://android-developers.googleblog.com/2019/03/changes-to-google-play-developer-api.html

### PR DESCRIPTION
**Pull request by scorpion35**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
- Fixes #14573
- Fixes #14271
- Updating AndroidPublisher to V3

### Description
- Updating `supply` to AndroidPublisherV3 (from v2)

#### Working/Fixed
- Release APK
  - With and without rollout
  - With uploading main and patch OBB
  - With referencing existing main and patch OBB
- Release bundle (AAB)
  - With and without rollout
- Update rollout of track
- Promote
  - Move release from one track to another
  - Also can update rollout
- Change log
  - Download
  - Upload
- Images
  - Download
  - Upload
- Screenshots
  - Download
  - Upload
- Metadata
  - Download
  - Upload
- Unit tests
- Actions
  - `download_from_play_store`
- Documentation
  - Updated migration notes

--- 

### How to use this PR
Updated your `Gemfile` to 👇 and run `bundle install`, `bundle update fastlane`, or `bundle update`
```
gem 'fastlane', :git => 'https://github.com/scorpion35/fastlane.git', :branch => 'androidpublisherv3'
```

### CODE REVIEW

**Comment by scorpion35** [comment]
I have signed using my both accounts :)

**Comment by ejkkan** [comment]
Any plans on getting this merged? :)

**Comment by joshdholtz** [comment]
@ejkkan I'm working on finishing this PR up for @scorpion35 right now 💪 Hopefully we can get this out soon ™️ 🤞

**Comment by joshdholtz** [comment]
@googlebot I consent.

File: supply/lib/supply.rb [hunk 1]
```diff
@@ -23,6 +24,13 @@ class << self
 
   CHANGELOGS_FOLDER_NAME = "changelogs"
 
+  # https://developers.google.com/android-publisher/#publishing
+  AVAILABLE_TRACKS = %w(production rollout beta alpha internal)
+  DEFAULT_TRACK = AVAILABLE_TRACKS[0]
+
+  # https://developers.google.com/android-publisher/api-ref/edits/tracks
+  RELEASE_STATUS = %w(completed draft halted inProgress)
```
Comment (joshdholtz) [inline supply/lib/supply.rb]: Move these statuses also to constants so we can use them other places without hardcoding strings

File: supply/lib/supply/client.rb [hunk 2]
```diff
@@ -236,6 +236,56 @@ def apk_listings(apk_version_code)
       end
     end
 
+    def release_listings(version)
+      ensure_active_edit!
+
+      # 'rollout' track returned by AndroidPublisherV3 doesn't have 'releases'.
```
Comment (joshdholtz) [inline supply/lib/supply/client.rb]: Figure out why `rollout` is referenced in this comment

File: supply/lib/supply/options.rb [hunk 3]
```diff
@@ -14,11 +13,39 @@ def self.available_options
...
+                                     description: "Version name",
+                                     code_gen_sensitive: true,
+                                     default_value: CredentialsManager::AppfileConfig.try_fetch_value(:version_name),
+                                     default_value_dynamic: true),
+        FastlaneCore::ConfigItem.new(key: :version_code,
+                                     env_name: "SUPPLY_VERSION_CODE",
+                                     short_option: "-C",
+                                     optional: true,
+                                     description: "Version code",
+                                     code_gen_sensitive: true,
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: false

File: supply/lib/supply/options.rb [hunk 4]
```diff
@@ -147,35 +174,41 @@ def self.available_options
                                      optional: true,
                                      description: "Whether to skip uploading APK",
                                      is_string: false,
-                                     default_value: false),
+                                     default_value: true),
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: Revert (breaking change)

File: supply/lib/supply/options.rb [hunk 5]
```diff
@@ -147,35 +174,41 @@ def self.available_options
...
         FastlaneCore::ConfigItem.new(key: :skip_upload_metadata,
                                      env_name: "SUPPLY_SKIP_UPLOAD_METADATA",
                                      optional: true,
-                                     description: "Whether to skip uploading metadata",
+                                     description: "Whether to skip uploading metadata, changelogs not included",
                                      is_string: false,
-                                     default_value: false),
+                                    default_value: true),
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: Revert (breaking change)

File: supply/lib/supply/options.rb [hunk 6]
```diff
@@ -147,35 +174,41 @@ def self.available_options
...
+        FastlaneCore::ConfigItem.new(key: :skip_upload_changelogs,
+                                     env_name: "SUPPLY_SKIP_UPLOAD_CHANGELOGS",
+                                     optional: true,
+                                     description: "Whether to skip uploading changelogs",
+                                     is_string: false,
+                                     default_value: true),
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: Maybe also make false? But maybe also just get rid of and use `skip_upload_metadata `

File: supply/lib/supply/options.rb [hunk 7]
```diff
@@ -147,35 +174,41 @@ def self.available_options
...
-                                     description: "Whether to skip uploading metadata",
+                                     description: "Whether to skip uploading metadata, changelogs not included",
                                      is_string: false,
-                                     default_value: false),
+                                    default_value: true),
+        FastlaneCore::ConfigItem.new(key: :skip_upload_changelogs,
+                                     env_name: "SUPPLY_SKIP_UPLOAD_CHANGELOGS",
+                                     optional: true,
+                                     description: "Whether to skip uploading changelogs",
+                                     is_string: false,
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: 💣

Comment (joshdholtz) [reply supply/lib/supply/options.rb]: Note to @joshdholtz: add `supply` into the rubocop rules for `is_string`

File: supply/lib/supply/uploader.rb [hunk 8]
```diff
@@ -60,17 +129,70 @@ def verify_config!
...
+        UI.user_error!("Track '#{Supply.config[:track]}' doesn't have any releases")
+      elsif releases.size > 1
+        UI.user_error!("Track '#{Supply.config[:track]}' has more than one release - use :version_code to filter the release to promote")
+      end
+
+      release = track_from.releases.first
+      track_to = @client.tracks(Supply.config[:track_promote_to]).first
+
+      if track_to
+        track_to.releases = [release]
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: Look into if we need to use `deactivate_on_promote` on `track_from` 👀

File: supply/lib/supply/options.rb [hunk 9]
```diff
@@ -227,16 +260,18 @@ def self.available_options
                                      env_name: "SUPPLY_DEACTIVATE_ON_PROMOTE",
                                      optional: true,
                                      description: "When promoting to a new track, deactivate the binary in the origin track",
+                                     deprecated: "No longer allowed with Android Publisher V3",
                                      is_string: false,
                                      default_value: true),
         FastlaneCore::ConfigItem.new(key: :version_codes_to_retain,
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: @joshdholtz TEST THIS PLEASE 🤔

File: supply/lib/supply/setup.rb [hunk 10]
```diff
@@ -12,14 +12,15 @@ def perform_download
 
       client.listings.each do |listing|
         store_metadata(listing)
-        create_screenshots_folder(listing)
```
Comment (joshdholtz) [inline supply/lib/supply/setup.rb]: @joshdholtz Look into why this is no longer needed

Comment (joshdholtz) [reply supply/lib/supply/setup.rb]: Scratch that. Its done in `download_images` with a `mkdir_p` 💡

File: supply/lib/supply/setup.rb [hunk 11]
```diff
@@ -43,56 +44,58 @@ def store_metadata(listing)
...
+      allowed_imagetypes = [Supply::IMAGES_TYPES, Supply::SCREENSHOT_TYPES].flatten
 
+      allowed_imagetypes.each do |image_type|
         begin
+          path = File.join(metadata_path, listing.language, IMAGES_FOLDER_NAME, image_type)
+
+          p = Pathname.new(path)
+          if IMAGES_TYPES.include?(image_type) # IMAGE_TYPES are stored in locale/images location
+            FileUtils.mkdir_p(p.dirname.to_s)
+          else # SCREENSHOT_TYPES go under their respective folders.
```
Comment (joshdholtz) [inline supply/lib/supply/setup.rb]:
```suggestion
          else # SCREENSHOT_TYPES are stored in locale/images/<screenshot_type>
```

File: supply/lib/supply/setup.rb [hunk 12]
```diff
@@ -43,56 +44,58 @@ def store_metadata(listing)
...
 
-          url = client.fetch_images(image_type: image_type, language: listing.language).last
-          next unless url
+          urls = client.fetch_images(image_type: image_type, language: listing.language)
+          next if urls.nil? || urls.empty?
+
+          image_counter = 1 # Used to prefix the downloaded files, so order is preserved.
+          urls.each do |url|
+            if IMAGES_TYPES.include?(image_type) # IMAGE_TYPES are stored in locale/images location
+              file_path = "#{path}.#{FastImage.type(url)}"
```
Comment (joshdholtz) [inline supply/lib/supply/setup.rb]: Use `File.join`

File: supply/lib/supply/uploader.rb [hunk 13]
```diff
@@ -7,46 +7,115 @@ def perform_upload
...
 
       apk_version_codes.concat(Supply.config[:version_codes_to_retain]) if Supply.config[:version_codes_to_retain]
 
+
       # Only update tracks if we have version codes
       # Updating a track with empty version codes can completely clear out a track
       update_track(apk_version_codes) unless apk_version_codes.empty?
 
+
+      if !Supply.config[:rollout].nil? && Supply.config[:version_code].to_s != "" && Supply.config[:track].to_s != ""
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: Clean this logic up - hard to follow

Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: Is `version_code` needed? Maybe error out in `update_rollout` if multiple versions and `:version_code` is needed

File: supply/lib/supply/uploader.rb [hunk 14]
```diff
@@ -7,46 +7,115 @@ def perform_upload
...
+          all_languages.each do |language|
+            next if language.start_with?('.') # e.g. . or .. or hidden folders
+            UI.message("Preparing to upload for language '#{language}'...")
+
+            listing = client.listing_for_language(language)
+            
+            upload_metadata(language, listing) unless Supply.config[:skip_upload_metadata]
+            upload_images(language) unless Supply.config[:skip_upload_images]
+            upload_screenshots(language) unless Supply.config[:skip_upload_screenshots]
+            release_notes << upload_changelog(language, release.name) unless Supply.config[:skip_upload_changelogs]
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: Still unsure if `skip_upload_changelogs` should just use `skip_upload_metadata`? 🤔

File: supply/lib/supply/uploader.rb [hunk 15]
```diff
@@ -7,46 +7,115 @@ def perform_upload
...
+          end
+
+          upload_changelogs(release_notes, release, track) unless Supply.config[:skip_upload_changelogs]
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]:
```suggestion
          upload_changelogs(release_notes, release, track) unless release_notes.empty?
```

File: supply/lib/supply/uploader.rb [hunk 16]
```diff
@@ -7,46 +7,115 @@ def perform_upload
...
+      version_codes = Supply.config
+      if Supply.config[:version_code]
+        releases = releases.select { |r| r.version_codes.include?(Supply.config[:version_code]) }
+      end
+
+      release = releases.first
+      if release
+        completed = Supply.config[:rollout].to_f == 1
+        release.send("user_fraction=", completed ? nil : Supply.config[:rollout])
+        release.send("status=", 'completed') if completed
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: Why are we using `.send`? Just call the method

File: supply/lib/supply/uploader.rb [hunk 17]
```diff
@@ -7,46 +7,115 @@ def perform_upload
...
+
+        track.releases.delete_if { |r| !r.version_codes.include?(Supply.config[:version_code]) } if completed
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: Look into this 👀 Can we have multiple full rollout releases in a track????? 😖

File: supply/lib/supply/uploader.rb [hunk 18]
```diff
@@ -60,17 +129,70 @@ def verify_config!
...
+          releases: [release]
+        )
+      end
+
+      client.update_track(Supply.config[:track_promote_to], track_to)
     end
 
-    def upload_changelogs(language)
+    def upload_changelog(language, version_name)
+      UI.user_error!("Cannot find changelog - no version name found from uploaded APk/AAB") unless version_name
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]:
```suggestion
      UI.user_error!("Cannot find changelog - please specify :version_name") unless version_name
```
Ehhhhhhhhhhh?????

Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: Look into why this could error and comment on it

File: supply/lib/supply/uploader.rb [hunk 19]
```diff
@@ -79,7 +201,7 @@ def upload_changelogs(language)
       end
     end
 
-    def upload_changelog(language, version_code)
+    def upload_changelog_DEPRECATED(language, version_code)
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: 💣  this

File: supply/lib/supply/uploader.rb [hunk 20]
```diff
@@ -164,12 +286,12 @@ def upload_bundles
         UI.message("Preparing aab at path '#{aab_path}' for upload...")
         bundle_version_code = client.upload_bundle(aab_path)
 
-        if metadata_path
-          all_languages.each do |language|
-            next if language.start_with?('.') # e.g. . or .. or hidden folders
-            upload_changelog(language, bundle_version_code)
-          end
-        end
+        # if metadata_path
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: 💣 this

File: supply/lib/supply/uploader.rb [hunk 21]
```diff
@@ -209,15 +332,17 @@ def upload_binary_data(apk_path)
 
         upload_obbs(apk_path, apk_version_code)
 
-        if metadata_path
-          all_languages.each do |language|
-            next if language.start_with?('.') # e.g. . or .. or hidden folders
-            upload_changelog(language, apk_version_code)
-          end
-        end
+        # if metadata_path
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: 💣 this

File: supply/lib/supply/uploader.rb [hunk 22]
```diff
@@ -230,14 +355,29 @@ def update_obb(apk_version_code, expansion_file_type, references_version, file_s
     end
 
     def update_track(apk_version_codes)
+      return if apk_version_codes.empty?
+
       UI.message("Updating track '#{Supply.config[:track]}'...")
       check_superseded_tracks(apk_version_codes) if Supply.config[:check_superseded_tracks]
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: @joshdholtz TEST THIS PLEASE

File: supply/lib/supply/uploader.rb [hunk 23]
```diff
@@ -230,14 +355,29 @@ def update_obb(apk_version_code, expansion_file_type, references_version, file_s
...
 
-      if Supply.config[:track].eql?("rollout")
-        client.update_track(Supply.config[:track], Supply.config[:rollout] || 0.1, apk_version_codes)
-      else
-        client.update_track(Supply.config[:track], 1.0, apk_version_codes)
+      track_release = AndroidPublisher::TrackRelease.new(
+        name: Supply.config[:version_name],
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: Do a check if empty string and set as nil if found

File: supply/lib/supply/uploader.rb [hunk 24]
```diff
@@ -230,14 +355,29 @@ def update_obb(apk_version_code, expansion_file_type, references_version, file_s
...
+        release_notes: [],
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: Either 💣  this line (if possible) or comment on why its empty

File: supply/lib/supply/uploader.rb [hunk 25]
```diff
@@ -230,14 +355,29 @@ def update_obb(apk_version_code, expansion_file_type, references_version, file_s
...
+        status: "completed",
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: Use constant

File: supply/lib/supply/uploader.rb [hunk 26]
```diff
@@ -230,14 +355,29 @@ def update_obb(apk_version_code, expansion_file_type, references_version, file_s
...
+        version_codes: apk_version_codes
+      )
+
+      if Supply.config[:rollout]
+        track_release.status = "inProgress"
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: Use constant

File: supply/lib/supply/uploader.rb [hunk 27]
```diff
@@ -230,14 +355,29 @@ def update_obb(apk_version_code, expansion_file_type, references_version, file_s
...
+        track_release.user_fraction = Supply.config[:rollout].to_f
       end
+
+      track = AndroidPublisher::Track.new(
```
Comment (joshdholtz) [inline supply/lib/supply/uploader.rb]: DONT CREATE A NEW TRACK - this will override the current track and its releases???? Fetch an existing track, possible keep releases if `version_codes_to_retain` is used, and then save

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Changes for @joshdholtz to make 🙃

File: supply/lib/supply/uploader.rb (hunk 14 above)
Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: I think we will keep it separate 👌

File: supply/lib/supply/uploader.rb (hunk 27 above)
Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: This was _actually_ okay because Google Play handles some things on their side but reworked anyway to feel less dangerous

File: supply/lib/supply/uploader.rb (hunk 8 above)
[the remark this answers is not in the document]
Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: Google play does this on its own

File: supply/lib/supply/uploader.rb (hunk 17 above)
Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: This is 👌 Added a comment why

File: supply/lib/supply/uploader.rb (hunk 24 above)
Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: All good without this 👌

File: supply/lib/supply/uploader.rb (hunk 22 above)
Comment (joshdholtz) [reply supply/lib/supply/uploader.rb]: Deleted this because Google Play does this itself now 
Screen Shot 2019-11-06 at 7 05 31 AM

File: fastlane.gemspec [hunk 28]
```diff
@@ -98,6 +98,9 @@ Gem::Specification.new do |spec|
   if Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.4')
     spec.add_dependency('signet', '<= 0.11') # Because yeah
     STDERR.puts("WARNING: Locking to a potentially insecure version of 'signet' because you are using a version of Ruby which is marked as End-Of-Life. Please upgrade your Ruby installation to 2.4 or later")
+
+    spec.add_dependency('google-cloud-env', '1.2.1')
+    spec.add_dependency('google-cloud-core', '1.3.2')
```
Comment (joshdholtz) [inline fastlane.gemspec]: Move this into a separate PR and then rebase

File: fastlane/lib/fastlane/actions/docs/upload_to_play_store.md [hunk 29]
```diff
@@ -75,7 +75,7 @@ This will also upload app metadata if you previously ran `fastlane supply init`.
 To gradually roll out a new build use
 
 ```no-highlight
-fastlane supply --apk path/app.apk --track rollout --rollout 0.5
+fastlane supply --apk path/app.apk --rollout 0.5
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/docs/upload_to_play_store.md]: Add track back in here

File: fastlane/lib/fastlane/actions/docs/upload_to_play_store.md [hunk 30]
```diff
@@ -110,7 +110,7 @@ This will also upload app metadata if you previously ran `fastlane supply init`.
 To gradually roll out a new build use
 
 ```no-highlight
-fastlane supply --aab path/app.aab --track rollout --rollout 0.5
+fastlane supply --aab path/app.aab --rollout 0.5
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/docs/upload_to_play_store.md]: Add track back in here

File: supply/androidpublisherv3_tracking.md [hunk 31]
```diff
@@ -0,0 +1,57 @@
+# Fastlane - AndroidPublisher V3 Migration
```
Comment (joshdholtz) [inline supply/androidpublisherv3_tracking.md]: Delete this whole file

File: supply/lib/supply/options.rb [hunk 32]
```diff
@@ -14,11 +13,36 @@ def self.available_options
                                      code_gen_sensitive: true,
                                      default_value: CredentialsManager::AppfileConfig.try_fetch_value(:package_name),
                                      default_value_dynamic: true),
+        FastlaneCore::ConfigItem.new(key: :version_name,
+                                     env_name: "SUPPLY_VERSION",
+                                     short_option: "-n",
+                                     optional: true,
+                                     description: "Version name (used when uploading new apks/aabs)",
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: Mention if not set, Google Play will default to version number

Comment (joshdholtz) [reply supply/lib/supply/options.rb]: Add this into migration doc

File: supply/lib/supply/options.rb [hunk 33]
```diff
@@ -14,11 +13,36 @@ def self.available_options
...
+                                     default_value: CredentialsManager::AppfileConfig.try_fetch_value(:version_name),
+                                     default_value_dynamic: true),
+        FastlaneCore::ConfigItem.new(key: :version_code,
+                                     env_name: "SUPPLY_VERSION_CODE",
+                                     short_option: "-C",
+                                     optional: true,
+                                     description: "Version code (used when updating rollout or promoting specific versions)",
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: Add this into migration doc (and maybe an example)

File: supply/lib/supply/options.rb [hunk 34]
```diff
@@ -14,11 +13,36 @@ def self.available_options
...
+                                     default_value: CredentialsManager::AppfileConfig.try_fetch_value(:version_code),
+                                     default_value_dynamic: true),
+        FastlaneCore::ConfigItem.new(key: :release_status,
+                                     env_name: "RELEASE_STATUS",
+                                     short_option: "-e",
+                                     optional: true,
+                                     description: "Release status (used when uploading new apks/aabs) - valid values are #{Supply::ReleaseStatus::ALL.join(', ')}",
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: Add to migration doc

File: supply/lib/supply/options.rb (hunk 7 above)
Comment (joshdholtz) [reply supply/lib/supply/options.rb]: Add to migration that this used to be under `skip_upload_meta`

File: supply/lib/supply/options.rb [hunk 35]
```diff
@@ -215,6 +247,7 @@ def self.available_options
                                      env_name: "SUPPLY_CHECK_SUPERSEDED_TRACKS",
                                      optional: true,
                                      description: "Check the other tracks for superseded versions and disable them",
+                                     deprecated: "Google Play does this automatically now",
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: Add to migration docs

File: supply/lib/supply/options.rb [hunk 36]
```diff
@@ -227,16 +260,18 @@ def self.available_options
                                      env_name: "SUPPLY_DEACTIVATE_ON_PROMOTE",
                                      optional: true,
                                      description: "When promoting to a new track, deactivate the binary in the origin track",
+                                     deprecated: "Google Play does this automatically now",
```
Comment (joshdholtz) [inline supply/lib/supply/options.rb]: Add to migration docs

**Review by joshdholtz — CHANGES_REQUESTED** [review]
